types/node/v10/tls.d.ts

```diff
@@ -156,6 +171,32 @@ declare module "tls" {
          * and the SSL/TLS protocol version of the current connection.
          */
         getCipher(): CipherNameAndProtocol;
+        /**
+         * Returns an object representing the type, name, and size of parameter
+         * of an ephemeral key exchange in Perfect Forward Secrecy on a client
+         * connection. It returns an empty object when the key exchange is not
+         * ephemeral. As this is only supported on a client socket; null is
+         * returned if called on a server socket. The supported types are 'DH'
+         * and 'ECDH'. The name property is available only when type is 'ECDH'.
+         *
+         * For example: { type: 'ECDH', name: 'prime256v1', size: 256 }.
+         */
+        getEphemeralKeyInfo(): EphemeralKeyInfo | {} | null;
```

Comment:
Why `{}`?

Reply:
Object can be empty. Function can return ie. `{ type: 'ECDH', name: 'prime256v1', size: 256 }` or `{}` or `null`. I think it is totally different than returning `{ type?: string, name?: string, size?: number }`.

Do you have any alternate syntax for "empty object" type?

Reply:
I think it works:

```ts
const keyInfo: EphemeralKeyInfo | {} | null = tlsSocket.getEphemeralKeyInfo();
if (!keyInfo) {
    console.log(keyInfo, "is null")
} else if (!("type" in keyInfo)) {
    console.log(keyInfo, "is {}")
} else {
    console.log(keyInfo.name);
    console.log(keyInfo.type);
    console.log(keyInfo.size);
}
```

but I can replace EphemeralKeyInfo to `{ type?: string, name?: string, size?: number }` if you think that it will simplify this case.